### PR TITLE
[Bug] [connector-spark-redis] Fix config param issues of spark redis sink

### LINKED ISSUE #1522: [Bug] [seatunnel-connector-spark-redis] default config as official doc said can't run correctly
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

Doc from [spark redis sink](https://seatunnel.apache.org/docs/spark/configuration/sink-plugins/Redis) says only "host" and "port" are required in config file. But I got this
`ERROR org.apache.seatunnel.command.BaseTaskExecuteCommand  - Plugin[org.apache.seatunnel.spark.sink.Redis] contains invalid config, error: No configuration setting found for key 'data_type'`
in my local dev env

### SeaTunnel Version

dev

### SeaTunnel Config

```conf
env {
  spark.app.name = "SeaTunnel"
  spark.executor.instances = 2
  spark.executor.cores = 1
  spark.executor.memory = "1g"
  spark.master = local
}

source {
  Fake {
    result_table_name = "my_dataset"
  }
}

transform {
}

sink {
  Console {}
  Redis {
    host = "localhost"
    port = 6379
  }
}
```

### Running Command

```shell
start in my local dev env
```

### Error Exception

```log
ERROR org.apache.seatunnel.command.BaseTaskExecuteCommand  - Plugin[org.apache.seatunnel.spark.sink.Redis] contains invalid config, error: No configuration setting found for key 'data_type'
```

### Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by DingPengfei**
## Purpose of this pull request
Fix config param issues of spark redis sink
this close #1522 

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in you PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/developement/NewLicenseGuide.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-connectors/seatunnel-connectors-spark/seatunnel-connector-spark-redis/src/main/scala/org/apache/seatunnel/spark/sink/Redis.scala [hunk 1]
```diff
@@ -38,7 +38,7 @@ class Redis extends SparkBatchSink with Logging {
     val redisConfigs = new RedisConfig(RedisEndpoint(
       host = config.getString(HOST),
       port = config.getInt(PORT),
-      auth = config.getString(AUTH),
+      auth = if (config.getIsNull(AUTH)) null else config.getString(AUTH),
```
Comment (Hisoka-X) [inline seatunnel-connectors/seatunnel-connectors-spark/seatunnel-connector-spark-redis/src/main/scala/org/apache/seatunnel/spark/sink/Redis.scala:41]: Already put it into DefaultConfig in line 75, So `AUTH` alway be not null, same as `DATA_TYPE`

Comment (DingPengfei) [reply seatunnel-connectors/seatunnel-connectors-spark/seatunnel-connector-spark-redis/src/main/scala/org/apache/seatunnel/spark/sink/Redis.scala:41]: `DEFAULT_AUTH` is null, `config.getString(DEFAULT_AUTH)` can get an error

Comment (Hisoka-X) [reply seatunnel-connectors/seatunnel-connectors-spark/seatunnel-connector-spark-redis/src/main/scala/org/apache/seatunnel/spark/sink/Redis.scala:41]: OK, I didn't notice.

**Comment by Hisoka-X** [comment]
LGTM

**Review by Hisoka-X — APPROVED** [review]

**Review by ruanwenjun — APPROVED** [review]
Good job, LGTM.

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**